### PR TITLE
fix: [SRM-10711]: Updated Prometheus connector payload

### PR DESCRIPTION
**Pull request by praneshg239**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by praneshg239** [comment]
retrigger coverage

**Comment by praneshg239** [comment]
trigger cypress

File: src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts [hunk 1]
```diff
@@ -1394,9 +1395,12 @@ export const buildPrometheusPayload = (formData: FormData) => {
       projectIdentifier: formData.projectIdentifier,
       orgIdentifier: formData.orgIdentifier,
       spec: {
+        ...transformStepHeadersAndParamsForPayloadForPrometheus(formData.headers),
         delegateSelectors: formData.delegateSelectors || {},
         url: formData.url,
-        accountId: formData.accountId
+        accountId: formData.accountId,
+        username: formData.username,
```
Comment (harshilbits) [inline src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts:1402]: NIT : Destructuring can be done here.

If its an urgent PR , can be picked in the next PR

**Review by harshilbits — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/PrometheusConnector/utils.ts [hunk 2]
```diff
@@ -5,11 +5,69 @@
...
 
-export function initializePrometheusConnectorWithStepData(
-  prevStepData?: ConnectorConfigDTO | null
-): ConnectorConfigDTO | undefined {
+async function transformSpecHeader(
+  scopeQueryParams: GetSecretV2QueryParams,
+  entity?: CustomHealthConnectorDTO['headers']
+): Promise<BaseCompFields['headers']> {
+  if (!entity?.length) {
+    return [
```
Comment (deepesh-ui) [inline src/modules/35-connectors/components/CreateConnector/PrometheusConnector/utils.ts:23]: can use predefined constant for this return value

File: src/modules/35-connectors/components/CreateConnector/PrometheusConnector/utils.ts [hunk 3]
```diff
@@ -5,11 +5,69 @@
...
+      } as PlainEntity)
+    }
+    return e.valueEncrypted
+  })
+
+  await Promise.all(
+    encryptedEntities.map(encryptedEntity =>
+      setSecretField(encryptedEntity.encryptedValueRef as string, scopeQueryParams)
+    )
+  ).then(secretObjs => {
```
Comment (deepesh-ui) [inline src/modules/35-connectors/components/CreateConnector/PrometheusConnector/utils.ts:52]: good to add a catch here

**Review by deepesh-ui — DISMISSED** [review]
approved with comments

File: src/modules/35-connectors/components/CreateConnector/PrometheusConnector/utils.ts (hunk 2 above)
Comment (praneshg239) [reply src/modules/35-connectors/components/CreateConnector/PrometheusConnector/utils.ts:23]: this we are going to use only here, so I thought of avoiding constant for this. Anyway, I'll add in next PR

**Review by deepesh-ui — APPROVED** [review]

**Review by harshilbits — DISMISSED** [review]

**Comment by praneshg239** [comment]
trigger cypress

**Pull request merged**